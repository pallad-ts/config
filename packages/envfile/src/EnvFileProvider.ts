import {Provider} from "@pallad/config";

export class EnvFileProvider extends Provider<string> {
    constructor(private key: string,
                private envFileVars: Record<string, string>) {
        super();
    }

    getDescription() {
        return `"${this.key}" from ENV file`;
    }

    isAvailable() {
        if (this.key in this.envFileVars) {
            return this.envFileVars[this.key] !== '';
        }
        return false;
    }

    protected retrieveValue() {
        return this.envFileVars[this.key];
    }
}
