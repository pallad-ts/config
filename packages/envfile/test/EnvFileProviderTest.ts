import {EnvFileProvider} from '@src/EnvFileProvider';

describe('EnvFileProvider', () => {
    describe('availability', () => {
        it('is available if ENV exists and is not empty', () => {
            const provider = new EnvFileProvider('FOO', {FOO: 'test'});

            expect(provider.isAvailable())
                .toBe(true);
        });

        it('not available if ENV does not exist', () => {
            const provider = new EnvFileProvider('FOO', {});

            expect(provider.isAvailable())
                .toBe(false);
        });

        it('not available if ENV exists but is empty', () => {
            const provider = new EnvFileProvider('FOO', {FOO: ''});

            expect(provider.isAvailable())
                .toBe(false);
        });
    });

    it('description', () => {
        expect(new EnvFileProvider('FOO', {}).getDescription())
            .toMatchSnapshot();
    });

    it('getting value', () => {
        expect(new EnvFileProvider('FOO', {FOO: 'bar'}).getValue())
            .toEqual('bar');
    });
});
