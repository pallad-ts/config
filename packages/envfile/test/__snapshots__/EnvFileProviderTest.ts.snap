// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnvFileProvider description 1`] = `"\\"FOO\\" from ENV file"`;
